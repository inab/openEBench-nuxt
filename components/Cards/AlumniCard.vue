<template>
	<div
		class="alumni-card"
		@mouseover="isHovered = true"
		@mouseleave="isHovered = false"
	>
		<v-card class="mx-auto fixed-height">
			<v-card-title class="name-container">{{ alumni.name }}</v-card-title>
			<v-card-subtitle class="subtitle-container">
				{{ formattedYears }}
				<div class="worked-on" @click="toggleExpand">
					<span
						class="worked-on-text"
						:class="{ 'fade-in': isHovered || isMobile }"
					>
						worked on
					</span>
					<v-icon class="arrow-icon">{{
						showRoles ? 'mdi-chevron-up' : 'mdi-chevron-down'
					}}</v-icon>
				</div>
			</v-card-subtitle>
			<v-divider class="mx-4 mb-1"></v-divider>
			<v-expand-transition>
				<div v-show="showRoles">
					<v-card-text class="scrollable-content">
						<p class="roles-text">{{ sortedandFormattedRoles }}</p>
					</v-card-text>
				</div>
			</v-expand-transition>
		</v-card>
	</div>
</template>

<script>
import sharedMethodsMixin from '@/mixins/sharedMethodsMixin.js';

export default {
	name: 'AlumniCard',
	mixins: [sharedMethodsMixin],
	props: {
		alumni: {
			type: Object,
			required: true,
		},
		isMobile: {
			type: Boolean,
			default: false,
		},
	},
	data() {
		return {
			showRoles: false,
			isHovered: false,
		};
	},
	computed: {
		formattedYears() {
			return `${this.alumni.startYear} - ${this.alumni.endYear}`;
		},
		sortedandFormattedRoles() {
			const sorted = this.alumni.roles.slice().sort();
			return sorted.join(', ');
		},
	},
	methods: {
		toggleExpand() {
			this.showRoles = !this.showRoles;
		},
	},
};
</script>

<style lang="scss" scoped>
@import '../../assets/sharedStyles.css';

.v-card-text {
	flex: 1 1 auto;
	overflow: hidden;
}

.scrollable-content {
	overflow-y: auto;
	max-height: 80px;
	padding-right: 16px;
}

.alumni-card:hover {
	box-shadow: 0 4px 20px rgba(0, 0, 0, 20%);
	transition: box-shadow 0.3s ease-in-out;
}

.subtitle-container {
	display: flex;
	align-items: center;
	justify-content: space-between;
}

.worked-on {
	display: flex;
	align-items: center;
	cursor: pointer;
}

.worked-on-text {
	font-size: 12px;
	font-weight: 300;
	text-transform: lowercase;
	margin-right: 4px;
	opacity: 0;
	transition: opacity 0.3s ease-in-out;
}

.fade-in {
	opacity: 1;
}

.arrow-icon {
	font-size: 16px;
	margin-top: 2px;
}

.roles-text {
	font-size: 14px !important;
}

/* Media query for mobile devices */
@media (max-width: 1080px) {
	.worked-on-text {
		opacity: 1 !important;
	}
}
</style>
