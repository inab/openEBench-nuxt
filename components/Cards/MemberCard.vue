<template>
	<v-card class="mx-auto my-4" :max-width="374">
		<!-- Centered container for circles -->
		<div class="circle-container">
			<!-- Outer circle -->
			<div class="outer-circle" style="margin-top: 10px">
				<!-- Inner circle with white background -->
				<div class="inner-circle">
					<!-- Circle image of team member -->

					<!-- Use v-avatar to display the member's image -->
					<v-avatar size="185" class="avatar-wrapper">
						<!-- Conditional rendering of image based on URL type -->

						<img
							:src="require(`@/static/members/images${member.image}`)"
							:alt="getMemberAltText(member)"
							:class="{ 'grayscale-image': !isMobile }"
							style="object-fit: cover; height: 100%; width: 100%"
						/>
					</v-avatar>
				</div>
			</div>
		</div>
		<v-card-title>{{ member.name }}</v-card-title>
		<v-card-subtitle class="subtitle-container">
			<div v-for="(institution, idx) in member.institution" :key="idx">
				<v-tooltip bottom>
					<template #activator="{ on }">
						<a
							href="#"
							class="institution-link"
							v-on="on"
							@click.prevent="navigateToLink(institution.trim())"
							>{{ institution.trim() }}</a
						>
						<!-- Display comma if not the last item -->
						<span v-if="idx < member.institution.length - 1">,&nbsp;</span>
					</template>
					<span>{{ getInstitutionFullName(institution).name }}</span>
				</v-tooltip>
			</div>
			<v-btn icon class="arrow" @click="toggleExpand(member)">
				<v-icon>{{
					member.show ? 'mdi-chevron-up' : 'mdi-chevron-down'
				}}</v-icon>
			</v-btn>
		</v-card-subtitle>

		<v-expand-transition>
			<div v-show="member.show">
				<v-divider class="mx-4"></v-divider>
				<v-card-text>
					<v-layout row wrap class="icon-row mt-2" justify-center>
						<v-flex v-for="(role, idx) in member.roles" :key="idx" xs4>
							<v-card
								v-if="!isMobile"
								class="cell"
								:style="{ backgroundColor: getItemColor(role.name) }"
								@mouseenter="role.hover = true"
								@mouseleave="role.hover = false"
							>
								<div class="content" :class="{ hovered: role.hover }">
									<div class="icon-container">
										<div
											:style="{
												backgroundImage: getItemIcon(role.name),
											}"
											class="icon"
										></div>
									</div>
									<div class="word">{{ formatRoleName(role.name) }}</div>
								</div>
							</v-card>
							<v-card
								v-if="isMobile"
								class="cell"
								:style="{ backgroundColor: getItemColor(role.name) }"
								@click="toggleRole(role)"
							>
								<div class="content" :class="{ hovered: role.hover }">
									<div class="icon-container">
										<div
											:style="{ backgroundImage: getItemIcon(role.name) }"
											class="icon"
										></div>
									</div>
									<div class="word">{{ formatRoleName(role.name) }}</div>
								</div>
							</v-card>
						</v-flex>
					</v-layout>
				</v-card-text>
			</div>
		</v-expand-transition>
		<v-divider class="mx-4 mb-1"></v-divider>

		<div class="px-4 mb-2 centered-chips">
			<v-chip-group>
				<v-chip
					v-if="member.orcid"
					class="chip"
					target="_blank"
					:href="member.orcid"
				>
					<img
						src="@/static/images/teams/orcid.svg"
						alt="ORCID Logo"
						class="logo chip-with-logo"
					/>ORCID
				</v-chip>
				<v-chip
					v-if="member.github"
					class="chip"
					target="_blank"
					:href="member.github"
				>
					<img
						src="@/static/images/teams/github-logo.png"
						alt="GitHub Logo"
						class="logo chip-with-logo"
					/>GITHUB
				</v-chip>
			</v-chip-group>
		</div>
	</v-card>
</template>

<script>
import sharedMethodsMixin from '@/mixins/sharedMethodsMixin.js';

export default {
	name: 'MemberCard',
	props: {
		member: {
			type: Object,
			required: true,
		},
		isMobile: {
			type: Boolean,
			default: false,
		},
		items: {
			type: Array,
			required: true,
		},
		institutionMapping: { type: Object, required: true },
	},
	mixins: [sharedMethodsMixin],
	methods: {
		getMemberAltText(member) {
			return `Photo of ${member.name}`;
		},
		navigateToLink(institution) {
			const institutionInfo = this.getInstitutionFullName(institution);
			if (institutionInfo.link) {
				window.open(institutionInfo.link, '_blank');
			}
		},
		getInstitutionFullName(acronym) {
			const institution = this.institutionMapping[acronym];
			if (institution) {
				return { name: institution.name, link: institution.link };
			}
			return { name: acronym, link: '#' };
		},
		toggleExpand(member) {
			member.show = !member.show;
		},
	},
};
</script>

<style scoped>
@import '../../assets/sharedStyles.css';

.chip-with-logo {
	margin-bottom: 2px;
}

.logo {
	margin-right: 5px;
	width: 20px;
	height: 20px;
}

.arrow {
	margin-left: auto;
}

.circle-container {
	display: flex;
	justify-content: center;
	align-items: center;
}

.outer-circle {
	width: 240px;
	height: 240px;
	border-radius: 50%;
	border: 10px solid #0b579f;
	display: flex;
	justify-content: center;
	align-items: center;
}

.inner-circle {
	width: 200px;
	height: 200px;
	border-radius: 50%;
	border: 10px solid white;
	display: flex;
	justify-content: center;
	align-items: center;
	background-color: white;
}

.grayscale-image {
	filter: grayscale(100%);
	transition: filter 0.3s ease;
}

.avatar-wrapper:hover .grayscale-image {
	filter: none;
}

.centered-chips {
	display: flex;
	justify-content: center;
}

.text {
	opacity: 0.6;
}

.institution-link {
	cursor: pointer;
	text-decoration: none;
	color: inherit;
	outline: none;
	white-space: nowrap;
}

.institution-link:hover {
	text-decoration: underline;
	color: #0b579f;
}
</style>
