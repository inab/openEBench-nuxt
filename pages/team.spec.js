import { shallowMount } from '@vue/test-utils';
import TeamPage from '@/pages/team.vue';
import membersData from '@/static/members/membersPage.json';
import MemberCard from '@/components/Cards/MemberCard.vue';
import AlumniCard from '@/components/Cards/AlumniCard.vue';

describe('TeamPage.vue', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallowMount(TeamPage, {
			mocks: {
				$config: {
					OEB_LEGACY_ANGULAR_URI: 'http://example.com',
				},
				$parent: {
					$emit: jest.fn(),
				},
			},
			data() {
				return {
					members: [],
					alumnis: [],
					isMobile: false,
					institutionMapping: {},
					rolesLegend: [],
				};
			},
			methods: {
				fetchData: jest.fn(),
			},
		});
		// Mock the mounted lifecycle hook to directly set the members data
		wrapper.setData({
			members: membersData.Members,
			alumnis: membersData.Alumni,
			institutionMapping: membersData.Institutions,
			rolesLegend: membersData.Roles.sort((a, b) =>
				a.name.localeCompare(b.name)
			),
		});
	});

	afterEach(() => {
		wrapper.destroy();
	});

	it('should include MemberCard and AlumniCard components', () => {
		expect(wrapper.findComponent(MemberCard).exists()).toBe(true);
		expect(wrapper.findComponent(AlumniCard).exists()).toBe(true);
	});

	it('should render the page title correctly', () => {
		expect(wrapper.find('h1').text()).toBe('The Team of OpenEBench');
	});

	it('should render the correct number of MemberCard components for leaders', () => {
		const leaders = wrapper.vm.leaders;
		const leaderCards = wrapper
			.findAllComponents(MemberCard)
			.filter((card) => leaders.includes(card.props().member));
		expect(leaderCards.length).toBe(leaders.length);
	});

	it('should render the correct number of MemberCard components for teammates', () => {
		const teammates = wrapper.vm.teammates;
		const teammateCards = wrapper
			.findAllComponents(MemberCard)
			.filter((card) => teammates.includes(card.props().member));
		expect(teammateCards.length).toBe(teammates.length);
	});

	it('should render the AlumniCard components for alumni', () => {
		const alumniCards = wrapper.findAllComponents(AlumniCard);
		const alumnis = wrapper.vm.alumnis;
		expect(alumniCards.length).toBe(alumnis.length);
	});

	it('should correctly compute leaders and teammates', () => {
		const leaders = wrapper.vm.leaders;
		const teammates = wrapper.vm.teammates;

		expect(leaders.every((member) => member.roles.includes('Leadership'))).toBe(
			true
		);
		expect(
			teammates.every((member) => !member.roles.includes('Leadership'))
		).toBe(true);
	});

	it('should check if the device is mobile', () => {
		wrapper.vm.checkMobile();
		expect(wrapper.vm.isMobile).toBe(window.innerWidth <= 850);
	});

	it('matches the snapshot', () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it('should initialize breadcrumbs correctly', () => {
		const breadcrumbs = wrapper.vm.breadcrumbs;
		expect(breadcrumbs).toEqual([
			{ text: 'Home', disabled: false, exact: true, to: '/' },
			{ text: 'About', disabled: true },
			{ text: 'Team', disabled: true },
		]);
	});

	it('should sort members by name', () => {
		const sortedMembers = wrapper.vm.sortedByName([
			{ name: 'Charlie' },
			{ name: 'Alice' },
			{ name: 'Bob' },
		]);
		expect(sortedMembers).toEqual([
			{ name: 'Alice' },
			{ name: 'Bob' },
			{ name: 'Charlie' },
		]);
	});
});
