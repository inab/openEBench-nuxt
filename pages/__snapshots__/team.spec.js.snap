// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamPage.vue matches the snapshot 1`] = `
"<v-container-stub tag=\\"div\\">
  <h1 class=\\"text--h3\\">The Team of OpenEBench</h1>
  <p>
    Here, you can get to know the current and
    <a href=\\"#alumni\\">past members</a> of the team behind the design,
    development and operation of OpenEBench. OpenEBench is the ELIXIR Open
    Data platform supporting community-driven scientific benchmarking
    activities and the technical monitoring of research software in Life
    Sciences.
  </p>
  <p>
    The OpenEBench team is a committed group of professionals with different
    roles and complementary expertise who work together to maintain OpenEBench
    and add new functionality for a better user experience. The team is mainly
    affiliated with the
    <a href=\\"https://bsc.es/\\" target=\\"_blank\\">Barcelona Supercomputing Center (BSC).<v-icon-stub small=\\"\\" right=\\"\\">mdi-open-in-new</v-icon-stub></a>
    However, we are open to collaborations worldwide as OpenEBench follows the
    Open Science principles since its inception to facilitate working
    together.
  </p>
  <p>
    The overall objective is to automatise the interactions within the
    platform as much as possible so users can become an integral part of its
    operations by curating and updating existing information within it.
  </p> <br>
  <h2 class=\\"text--h3 section-title-border\\">Leaders</h2>
  <v-row-stub tag=\\"div\\">
    <v-col-stub cols=\\"12\\" sm=\\"6\\" md=\\"4\\" tag=\\"div\\">
      <membercard-stub member=\\"[object Object]\\" items=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" institutionmapping=\\"[object Object]\\"></membercard-stub>
    </v-col-stub>
    <v-col-stub cols=\\"12\\" sm=\\"6\\" md=\\"4\\" tag=\\"div\\">
      <membercard-stub member=\\"[object Object]\\" items=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" institutionmapping=\\"[object Object]\\"></membercard-stub>
    </v-col-stub>
    <v-col-stub cols=\\"12\\" sm=\\"6\\" md=\\"4\\" tag=\\"div\\">
      <membercard-stub member=\\"[object Object]\\" items=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" institutionmapping=\\"[object Object]\\"></membercard-stub>
    </v-col-stub>
  </v-row-stub>
  <h2 class=\\"section-title-border mt-2\\">Group Members</h2>
  <v-row-stub tag=\\"div\\">
    <v-col-stub cols=\\"12\\" sm=\\"6\\" md=\\"4\\" tag=\\"div\\">
      <membercard-stub member=\\"[object Object]\\" items=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" institutionmapping=\\"[object Object]\\"></membercard-stub>
    </v-col-stub>
    <v-col-stub cols=\\"12\\" sm=\\"6\\" md=\\"4\\" tag=\\"div\\">
      <membercard-stub member=\\"[object Object]\\" items=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" institutionmapping=\\"[object Object]\\"></membercard-stub>
    </v-col-stub>
    <v-col-stub cols=\\"12\\" sm=\\"6\\" md=\\"4\\" tag=\\"div\\">
      <membercard-stub member=\\"[object Object]\\" items=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" institutionmapping=\\"[object Object]\\"></membercard-stub>
    </v-col-stub>
    <v-col-stub cols=\\"12\\" sm=\\"6\\" md=\\"4\\" tag=\\"div\\">
      <membercard-stub member=\\"[object Object]\\" items=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" institutionmapping=\\"[object Object]\\"></membercard-stub>
    </v-col-stub>
    <v-col-stub cols=\\"12\\" sm=\\"6\\" md=\\"4\\" tag=\\"div\\">
      <membercard-stub member=\\"[object Object]\\" items=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" institutionmapping=\\"[object Object]\\"></membercard-stub>
    </v-col-stub>
    <v-col-stub cols=\\"12\\" sm=\\"6\\" md=\\"4\\" tag=\\"div\\">
      <membercard-stub member=\\"[object Object]\\" items=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" institutionmapping=\\"[object Object]\\"></membercard-stub>
    </v-col-stub>
    <v-col-stub cols=\\"12\\" sm=\\"6\\" md=\\"4\\" tag=\\"div\\">
      <membercard-stub member=\\"[object Object]\\" items=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" institutionmapping=\\"[object Object]\\"></membercard-stub>
    </v-col-stub>
    <v-col-stub cols=\\"12\\" sm=\\"6\\" md=\\"4\\" tag=\\"div\\">
      <membercard-stub member=\\"[object Object]\\" items=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" institutionmapping=\\"[object Object]\\"></membercard-stub>
    </v-col-stub>
    <v-col-stub cols=\\"12\\" sm=\\"6\\" md=\\"4\\" tag=\\"div\\">
      <membercard-stub member=\\"[object Object]\\" items=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" institutionmapping=\\"[object Object]\\"></membercard-stub>
    </v-col-stub>
    <v-col-stub cols=\\"12\\" sm=\\"6\\" md=\\"4\\" tag=\\"div\\">
      <membercard-stub member=\\"[object Object]\\" items=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" institutionmapping=\\"[object Object]\\"></membercard-stub>
    </v-col-stub>
  </v-row-stub> <br> <br id=\\"alumni\\"> <br>
  <v-divider-stub></v-divider-stub>
  <h2 class=\\"mt-2\\">The OpenEBench Alumni network</h2>
  <p>
    Different professionals have contributed to the development of OpenEBench
    since its inception in 2017. In this section, you can find past members
    and their main roles within the team. We are grateful for their
    contributions to the platform and interactions with different communities.
  </p> <br>
  <v-row-stub tag=\\"div\\">
    <v-col-stub cols=\\"12\\" sm=\\"6\\" md=\\"4\\" tag=\\"div\\">
      <alumnicard-stub alumni=\\"[object Object]\\" items=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" institutionmapping=\\"[object Object]\\"></alumnicard-stub>
    </v-col-stub>
    <v-col-stub cols=\\"12\\" sm=\\"6\\" md=\\"4\\" tag=\\"div\\">
      <alumnicard-stub alumni=\\"[object Object]\\" items=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" institutionmapping=\\"[object Object]\\"></alumnicard-stub>
    </v-col-stub>
    <v-col-stub cols=\\"12\\" sm=\\"6\\" md=\\"4\\" tag=\\"div\\">
      <alumnicard-stub alumni=\\"[object Object]\\" items=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" institutionmapping=\\"[object Object]\\"></alumnicard-stub>
    </v-col-stub>
    <v-col-stub cols=\\"12\\" sm=\\"6\\" md=\\"4\\" tag=\\"div\\">
      <alumnicard-stub alumni=\\"[object Object]\\" items=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" institutionmapping=\\"[object Object]\\"></alumnicard-stub>
    </v-col-stub>
    <v-col-stub cols=\\"12\\" sm=\\"6\\" md=\\"4\\" tag=\\"div\\">
      <alumnicard-stub alumni=\\"[object Object]\\" items=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" institutionmapping=\\"[object Object]\\"></alumnicard-stub>
    </v-col-stub>
    <v-col-stub cols=\\"12\\" sm=\\"6\\" md=\\"4\\" tag=\\"div\\">
      <alumnicard-stub alumni=\\"[object Object]\\" items=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" institutionmapping=\\"[object Object]\\"></alumnicard-stub>
    </v-col-stub>
    <v-col-stub cols=\\"12\\" sm=\\"6\\" md=\\"4\\" tag=\\"div\\">
      <alumnicard-stub alumni=\\"[object Object]\\" items=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" institutionmapping=\\"[object Object]\\"></alumnicard-stub>
    </v-col-stub>
    <v-col-stub cols=\\"12\\" sm=\\"6\\" md=\\"4\\" tag=\\"div\\">
      <alumnicard-stub alumni=\\"[object Object]\\" items=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" institutionmapping=\\"[object Object]\\"></alumnicard-stub>
    </v-col-stub>
  </v-row-stub>
</v-container-stub>"
`;
